tests/services/test_alert_rules.py

```diff
@@ -27,6 +33,54 @@ def mock_detection_list_user_service(mocker, py42_response):
     return detection_list_user_service
 
 
+@pytest.fixture
+def mock_detection_list_get_by_id_failure_when_invalid_id(
+    mocker, mock_connection, py42_response
+):
+    response = mocker.MagicMock(spec=Response)
+    response.status_code = 400
+    exception = mocker.MagicMock(spec=Py42BadRequestError)
+    exception.response = response
+    detection_list_user_service = mocker.MagicMock(spec=DetectionListUserService)
+    detection_list_user_service.create_if_not_exists.return_value = False
+    detection_list_user_service.get_by_id.side_effect = Py42BadRequestError(
+        exception, ""
+    )
+    return detection_list_user_service
+
+
+@pytest.fixture
+def mock_detection_list_post_failure_when_invalid_rule_id(
+    mocker, mock_connection, py42_response
+):
+    response = mocker.MagicMock(spec=Response)
+    response.status_code = 400
+    exception = mocker.MagicMock(spec=Py42NotFoundError)
+    exception.response = response
+    mock_connection.post.side_effect = Py42NotFoundError(exception, "")
+    detection_list_user_service = mocker.MagicMock(spec=DetectionListUserService)
+    detection_list_user_service.create_if_not_exists.return_value = True
+    detection_list_user_service.get_by_id.return_value = py42_response
+    return detection_list_user_service
+
+
+@pytest.fixture
+def mock_detection_list_user_service_create_user(
+    mocker, py42_response, http_error, mock_connection
+):
+    py42_response.text = MOCK_DETECTION_LIST_GET_RESPONSE
+    py42_response.data = json.loads(MOCK_DETECTION_LIST_GET_RESPONSE)
+    response = mocker.MagicMock(spec=Response)
+    response.status_code = 400
+    exception = mocker.MagicMock(spec=Py42BadRequestError)
+    exception.response = response
+    detection_list_user_service = mocker.MagicMock(spec=DetectionListUserService)
+    detection_list_user_service.get_by_id.return_value = py42_response
```

Comment:
Need to make this a side-effect as it gets called twice, first time it should fail with `BadRequest` error and then after the user is added by `create_user` it  should return valid response.